ngi_visualizations/fpkm_scatter/fpkm_scatter.py

```diff
@@ -213,13 +263,100 @@ def plot_fpkm_scatter (sample_1, sample_2, x_lab, y_lab, output_fn=False, linear
     logging.info("Saving to {} and {}".format(png_fn, pdf_fn))
     plt.savefig(png_fn)
     plt.savefig(pdf_fn)
+    
+    # Return the filenames and R2
+    return {'png': png_fn, 'pdf': pdf_fn, output_fn:r_squared }
+
 
-    # Return the filenames
-    return {'png': png_fn, 'pdf': pdf_fn}
+def make_heatmap(data):
+    """
+    Takes a dict of r2 values and generates a heatmap using matplotlib
+    """
+    
+    #list the sample names
+    names=set()
+    for i in data.keys():
+        names.update(i.split('-'))
+    names=sorted(list(names))
+    
+    #If the sample looks like it's form a NGI project, do specific cleaning
+    ngi = re.compile(r'^P\d+_\d+')
+    clean_names=[]
+    for i in names:
+        i=i.split(".")[0]   #Remove everything after the first dot
+        j=re.search(ngi, i)
+        if j:
+            if i.endswith('Aligned'):
+                  i = i[:-7]
+        clean_names.append(i)
+    
+    matrix = []
+    #Generate the data matrix (list of lists) 
+    for idx, x in enumerate(names):
+        matrix.append([])
+        for idy, y in enumerate(names):
+        	v = 0
+        	try:
+        		v = data["{}-{}".format(x,y)]
+        	except KeyError:
+        		try:
+        			v = data["{}-{}".format(y,x)]
+        		except KeyError:
+        			if x == y:
+        				v = 1
+        	matrix[idx].append(v)
 
+    ### Turn data into numpy aray:
+    data = np.array(matrix)
+   
+    fig, ax = plt.subplots()
+    
+    #set size
+    heatmap = ax.pcolor(data, cmap='YlOrRd', vmin=0, vmax=1)
+    fig = plt.gcf()
+    ax.set_frame_on(False)
+    
+    #make the plot square
+    plt.gca().set_aspect('equal', adjustable='box')
+    
+    # put the major ticks at the middle of each cell
+    ax.set_yticks(np.arange(data.shape[0]) +0.5 , minor=False)
+    ax.set_xticks(np.arange(data.shape[1]) +0.5 , minor=False)
+    
+    # want a more natural, table-like display
+    ax.invert_yaxis()
+    #ax.xaxis.tick_top()
+    
+    #adjust plot position to make room for longer sample names:
+    plt.subplots_adjust(bottom=0.4, right=None, left=0.3, top = None)
+    
+    plt.colorbar(heatmap, ax=ax) 
 
+    # Set the labels
+    labels = clean_names
+    ax.set_xticklabels(labels, minor=False)
+    ax.set_yticklabels(labels, minor=False)
     
+    # rotate the plot
+    plt.xticks(rotation=90)
+
+    ax.grid(False)
+    # Turn off all the ticks
+    ax = plt.gca()
+   
+    plt.title("Heatmap of R2 values")
+    #remove small ticks
+    for t in ax.xaxis.get_major_ticks():
+        t.tick1On = False
+        t.tick2On = False
+    for t in ax.yaxis.get_major_ticks():
+        t.tick1On = False
+        t.tick2On = False
+    logging.info("Saving heatmap to heatmap.png")
```

Comment:
For added value, you could add a new command line option to make it possible to customise this.